Modules/fact_manager.py

```diff
@@ -0,0 +1,336 @@
+        mecha.
+
+        Args:
+            fact_name: Name of fact this applies to.
+            fact_lang: langID of fact this applies to.
+            author: the user that triggered the log entry.
+            msg: Message to be written to log.
+            new_field: Contents to place in the 'new' field for the log row.
+            old_field: Contents to place in the 'old' field for the log row.
+
+        Returns: Nothing.
+        """
+        log_query = sql.SQL(f"INSERT INTO {self._FACT_LOG} VALUES "
+                            f"(DEFAULT, %s, %s, %s, %s, %s, %s, %s)")
+
+        query_data = (fact_name, fact_lang, author, msg, old_field, new_field,
+                      datetime.datetime.utcnow())
+
+        try:
+            await self.query(log_query, query_data)
+        except (psycopg2.ProgrammingError, psycopg2.DatabaseError):
+            log.exception("Unable to write transaction log to table.")
+            raise
+
+    async def mfd(self, name: str, lang: str) -> bool:
+        """
+        Toggles the MFD flag on a fact.
+
+        >>> mfd_value = await self.mfd("test", "en")
+
+        Args:
+            name: name of fact to update
+            lang: lang of fact to update
+
+        Returns: True/False that fact was set to.
+        """
+
+        query = sql.SQL(f"SELECT mfd FROM "
+                        f"{self._FACT_TABLE} WHERE name=%s AND lang=%s")
+
+        mfd_query = sql.SQL(f"UPDATE {self._FACT_TABLE} "
+                            f"SET mfd=%s WHERE name=%s AND lang=%s")
+
+        try:
+            result = await self.query(query, (name, lang))
+
+            # Invert MFD field value, and set it again.
+            mfd_value = not result[0][0]
+            await self.query(mfd_query, (mfd_value, name, lang))
+
+        except (psycopg2.ProgrammingError, psycopg2.DatabaseError) as error:
+            # ProgrammingError is a query failure, DatabaseError is database unavailable.
+            log.exception(f"Error setting MFD field value for {name}-{lang}")
+            raise error
+
+        return mfd_value
+
+    async def mfd_list(self, num_results=5) -> list:
+        """
+        Returns a list of facts marked for deletion
+
+        >>> mfd_list = await self.mfd_list(3)
+
+        Returns: list of facts
+        """
+        query = sql.SQL(f"SELECT name, lang FROM "
+                        f"{self._FACT_TABLE} WHERE mfd=%s "
+                        f"ORDER BY edited DESC LIMIT {num_results}")
+
+        try:
+            raw_results = await self.query(query, (True,))
+        except (psycopg2.ProgrammingError, psycopg2.DatabaseError) as error:
+            # ProgrammingError is a query failure, DatabaseError is database unavailable.
+            log.exception(f"Error getting MFD list.")
+            raise error
+        else:
+            result = [f"{item[0]}-{item[1]}" for item in raw_results]
+
+        return result
```

Comment:
Incorrect.  ``self.query()`` always returns something.  worst case you'd get is any empty list.

Reply:
Except `result` is only set by the `else` block. The query's return value is stored in `raw_results`.

Reply:
because the else: statement will only resolve if the try: block doesn't raise.  If an error is raised, then there won't be any assignment to result anyway.

Reply:
Exactly, which means that `result` will be `None` instead of an empty list.

Reply:
Aaand..you're absolutely right, and I'm the blind one. Rescinded.